servo/servo.py

```diff
@@ -435,6 +479,50 @@ def top_level_schema_json(self, *, all: bool = False) -> str:
             default=pydantic.json.pydantic_encoder,
         )
 
+    async def connect(self) -> None:
+        # must define dynamic function to allow configurable backoff
+        await backoff.on_exception(
+            backoff.expo,
+            httpx.HTTPError,
+            max_time=lambda: self.config.settings.backoff.max_time(),
+            max_tries=lambda: self.config.settings.backoff.max_tries(),
+            on_giveup=lambda details: self.logger.critical(
+                f"retries exhausted after {details['elapsed']} seconds, giving up"
+            ),
+        )(self.try_connect)()
+
+    async def try_connect(self) -> None:
+        self.logger.info("Saying HELLO.", end=" ")
+        try:
+            await self.post_event(
+                servo.api.Events.hello,
+                dict(
+                    agent=servo.api.user_agent(),
+                    telemetry=self.telemetry.values,
+                ),
+            )
+        except httpx.HTTPStatusError as error:
+            if (
```

Comment:
Given that you have a backoff() decorator  on this, it may be worth it configuring the backoff not to retry on 410
(this is cosmetic, won't affect operation very much - add the exclusion only if it is easy to do so)

Reply:
Deferring this work for a future PR